src/audio/mux/mux.c

```diff
@@ -200,7 +204,195 @@ static struct comp_dev *mux_new(const struct comp_driver *drv,
+		sink_c = buffer_acquire(sink);
+		audio_stream_init_alignment_constants(byte_align, frame_align_req,
+				__sparse_cache &sink_c->stream);
+
+		if (!sink_c->hw_params_configured) {
+			struct ipc4_audio_format out_fmt;
+
+			out_fmt = cd->md.output_format;
+			sink_c->stream.channels = out_fmt.channels_count;
+			sink_c->stream.rate = out_fmt.sampling_frequency;
+			audio_stream_fmt_conversion(out_fmt.depth,
+							out_fmt.valid_bit_depth,
+							&sink_c->stream.frame_fmt,
+							&sink_c->stream.valid_sample_fmt,
+							out_fmt.s_type);
+
+			sink_c->buffer_fmt = out_fmt.interleaving_style;
+			params->frame_fmt = sink_c->stream.frame_fmt;
+
+			for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
+				sink_c->chmap[i] = (out_fmt.ch_map >> i * 4) & 0xf;
+
+			sink_c->hw_params_configured = true;
+		}
+		buffer_release(sink_c);
+	}
+
+
+	if (!list_is_empty(&dev->bsource_list)) {
+		list_for_item(source_list, &dev->bsource_list)
+		{
+			source = container_of(source_list, struct comp_buffer, sink_list);
+			source_c = buffer_acquire(source);
+			audio_stream_init_alignment_constants(byte_align, frame_align_req,
+					__sparse_cache &source_c->stream);
+			j = source_c->id;
+			cd->config.streams[j].pipeline_id = source_c->pipeline_id;
+			if (j == 0) {
+				source_c->stream.channels =
+						cd->md.base_cfg.audio_fmt.channels_count;
+				source_c->stream.rate =
+						cd->md.base_cfg.audio_fmt.sampling_frequency;
+				audio_stream_fmt_conversion(cd->md.base_cfg.audio_fmt.depth,
+						cd->md.base_cfg.audio_fmt.valid_bit_depth,
+						&source_c->stream.frame_fmt,
+						&source_c->stream.valid_sample_fmt,
+						cd->md.base_cfg.audio_fmt.s_type);
+
+				source_c->buffer_fmt = cd->md.base_cfg.audio_fmt.interleaving_style;
+
+				for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
+					source_c->chmap[i] =
+						(cd->md.base_cfg.audio_fmt.ch_map >> i * 4) & 0xf;
+			} else {
+				source_c->stream.channels =
+						cd->md.reference_format.channels_count;
+				source_c->stream.rate = cd->md.reference_format.sampling_frequency;
+				audio_stream_fmt_conversion(cd->md.reference_format.depth,
+						cd->md.reference_format.valid_bit_depth,
+						&source_c->stream.frame_fmt,
+						&source_c->stream.valid_sample_fmt,
+						cd->md.reference_format.s_type);
+				source_c->buffer_fmt = cd->md.reference_format.interleaving_style;
+
+				for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
+					source_c->chmap[i] =
+						(cd->md.reference_format.ch_map >> i * 4) & 0xf;
+			}
+			source_c->hw_params_configured = true;
+			buffer_release(source_c);
+		}
+	}
+	mux_prepare_look_up_table(dev);
+}
+
+static int mux_get_attribute(struct comp_dev *dev, uint32_t type, void *value)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+
+	switch (type) {
+	case COMP_ATTR_BASE_CONFIG:
+		*(struct ipc4_base_module_cfg *)value = cd->md.base_cfg;
+		break;
+	default:
+		return -EINVAL;
+	}
+	return 0;
+}
+#endif
 static void mux_free(struct comp_dev *dev)
```

Comment:
Add empty line after L395.

Reply:
yes, an empty line after `#endif` would be good